presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/ShardCleaner.java

```diff
@@ -275,9 +267,35 @@ private synchronized void runBackupCleanup()
         }
     }
 
+    private synchronized void waitJitterTime()
+    {
+        try {
+            // jitter to avoid overloading database
+            long interval = this.localCleanerInterval.roundTo(SECONDS);
+            SECONDS.sleep(ThreadLocalRandom.current().nextLong(1, interval));
+        }
+        catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    private synchronized void runLocalCleanupAtStartup()
+    {
+        try {
+            Set<UUID> local = getLocalShards();
+            waitJitterTime();
+            cleanLocalShardsAtStartup(local);
+        }
+        catch (Throwable t) {
+            log.error(t, "Error cleaning local shards");
+            localJobErrors.update(1);
+        }
+    }
+
     private synchronized void runLocalCleanup()
     {
         try {
+            waitJitterTime();
```

Comment:
this method is exported to be called from jmx. Your change will cause it to sleep before actual invocation.